Tasks/Common/docker-common/pipelineutils.ts

```diff
@@ -19,25 +19,31 @@ function addLabel(hostName: string, labelName: string, variableName: string, lab
     }
 }
 
-function addCommonLabels(hostName: string, labels: string[]): void {
+function addCommonLabels(hostName: string, labels: string[], addPipelineData?: boolean): void {
     addLabel(hostName, "system.teamfoundationcollectionuri", "SYSTEM_TEAMFOUNDATIONCOLLECTIONURI", labels);
-    addLabel(hostName, "system.teamproject", "SYSTEM_TEAMPROJECT", labels);
-    addLabel(hostName, "build.repository.name", "BUILD_REPOSITORY_NAME", labels);
+    if (addPipelineData) {
+        addLabel(hostName, "system.teamproject", "SYSTEM_TEAMPROJECT", labels);
+        addLabel(hostName, "build.repository.name", "BUILD_REPOSITORY_NAME", labels);
+    }
 }
 
-function addBuildLabels(hostName: string, labels: string[]): void {
-    addLabel(hostName, "build.repository.uri", "BUILD_REPOSITORY_URI", labels);
-    addLabel(hostName, "build.sourcebranchname", "BUILD_SOURCEBRANCHNAME", labels);
+function addBuildLabels(hostName: string, labels: string[], addPipelineData?: boolean): void {
     addLabel(hostName, "build.sourceversion", "BUILD_SOURCEVERSION", labels);
```

Comment:
Any reason why this isn't inside the `if..else` block?

Reply:
It is that we would be adding few labels by default and would not add labels with PII if user opts out